src/index.ts

```diff
@@ -345,6 +350,37 @@ class Spanner extends GrpcService {
       [CLOUD_RESOURCE_HEADER]: this.projectFormattedName_,
     };
     this.directedReadOptions = directedReadOptions;
+    this._instance_admin_api = null;
+    this._database_admin_api = null;
+    this.emulatorHost = emulatorHost;
+  }
+
+  // Helper for session-related instance admin client API calls
+  instance_admin_api(this): void {
```

Comment:
Is this the method that a user should call in order to get an instance admin client? Or is it an internal helper method?

If it's an internal helper method: It should start with an underscore to indicate that it is a private method not intended for public use.

If it's the public method that users should call in order to get an admin client: Then it needs more documentation on what it does, and what users are expected to do with the return value. This function creates a new instance each time that it is called, which is a resource-heavy operation. The user is also responsible for closing the instance when they are done with it, which is different from the current situation where all (admin) clients are managed by the client library, and automatically closed when you close this `Spanner` instance. As it is a resource-heavy operation, we should also document that it could make sense for users to cache the instance that is returned by this method, and reuse it if they intend to execute multiple admin operations.
I would also suggest to rename the method to `create_instance_admin_client`, so it is cleared to the caller that it actually creates a new client.

Reply:
Sorry, I just realized that I scanned through this method too quickly (and I'm reviewing too many different versions of this PR in different languages). This implementation does __not__ always create a new instance. It only creates a new instance when it has not yet been created before.

The problem here is that it is not clear to the user what they should do with the client that is being returned:
- Is the client managed by the client library, and will the client library close it when the Spanner client is closed?
- Is the client managed by the user, and should they close it when they are done with it?

Currently, none of the above strategies are implemented in this solution. You should make a clear choice between one of the following:
1. The client is managed by the client library. That then also means that the client that you create here, should be closed when the `Spanner` client is closed. You should then also clearly document this function that the client that is being returned is a shared instance, and that the user should __never__ close the client that is returned themselves.
2. The client is managed by the user. That then means that you should never return a shared instance (so you should create a new one every time), and that the method should be clearly documented that the user is responsible for closing the client that is returned.

Reply:
After taking a look at the code my observation was that the management of the client is handled at the user end. Moreover, as there exists no direct correlation between the `Spanner` client and the client obtained from the `create_instance_admin_client()` and `create_database_admin_client()` methods, closing the` Spanner` client will not automatically close the `Admin` client. Additionally, in the existing samples, where hand-written code is utilized, there is no explicit handling for client closure.

Reply:
Based on offline discussions, I have updated the methods to add the created admin clients into the `clients_ `map. Now, closing the `spanner` client will also close the `admin` client that will get created by calling `get_instance_admin_client()` and `get_database_admin_client()` methods